ecr-login/cli/docker-credential-ecr-login/main.go

```diff
@@ -24,5 +27,66 @@ import (
 func main() {
 	defer log.Flush()
 	config.SetupLogger()
-	credentials.Serve(ecr.ECRHelper{ClientFactory: api.DefaultClientFactory{}})
+	helper := ecr.ECRHelper{ClientFactory: api.DefaultClientFactory{}}
+	if len(os.Args) > 1 && os.Args[1] == "eval" {
+		evalCommand(helper)
+	} else {
+		credentials.Serve(helper)
+	}
+}
+
+func evalCommand(helper credentials.Helper) {
+	server, email, passwordStdin, err := parseArgs(helper)
+	if err == nil {
+		var user, token string
+		user, token, err = helper.Get(server)
+		if err == nil {
+			var emailOpt string
+			if email {
+				emailOpt = " -e none"
+			}
+			var echoPassword, passwordOpt string
+			if passwordStdin {
+				echoPassword = fmt.Sprintf("echo %s|", token)
```

Comment:
It might be better to do this as `<<< %s` so that `echo`'s command line entry in `/proc` doesn't have the token.

Reply:
Fish does not support herestrings. So I am hesitant in making this change.